pkg/devconsole/proxy/proxy_test.go

```diff
@@ -0,0 +1,59 @@
+package proxy
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestProxyEndpoint(t *testing.T) {
+	tests := []struct {
+		testName string
+		method   string
+		expected map[string]interface{}
+	}{
+		{
+			testName: "valid GET",
+			method:   "GET",
+			expected: map[string]interface{}{
+				"statusCode": http.StatusOK,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+
+			server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+				rw.Header().Add("Content-Type", "application/json")
+				rw.WriteHeader(http.StatusOK)
+			}))
+			defer server.Close()
+
+			requestBody := map[string]interface{}{
+				"url":    server.URL,
+				"method": tt.method,
+			}
+			body, err := json.Marshal(requestBody)
+			if err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			}
+
+			req, err := http.NewRequest(http.MethodPost, "/proxy", bytes.NewBuffer(body))
+			if err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			}
+
+			actual, err := serve(req)
+			if err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			}
+
+			assert.Equal(t, tt.expected["statusCode"], actual.StatusCode)
```

Comment:
When you use the full types above you can compare the full response here...

```suggestion
      if !reflect.DeepEqual(tt.expectedResponse, actual) {
          t.Errorf("Response does not match expectation:\n%v\nbut got\n%v", tt.expectedResponse, actual);
      }
```